x-pack/legacy/plugins/reporting/server/lib/level_logger.ts

```diff
@@ -4,48 +4,46 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { ServerFacade } from '../../types';
+import { LoggerFactory } from 'src/core/server';
 
 const trimStr = (toTrim: string) => {
   return typeof toTrim === 'string' ? toTrim.trim() : toTrim;
 };
 
 export class LevelLogger {
-  private _logger: any;
+  private _logger: LoggerFactory;
   private _tags: string[];
+  public warning: (msg: string, tags?: string[]) => void;
 
-  public warn: (msg: string, tags?: string[]) => void;
-
-  static createForServer(server: ServerFacade, tags: string[]) {
-    const serverLog: ServerFacade['log'] = (tgs: string[], msg: string) => server.log(tgs, msg);
-    return new LevelLogger(serverLog, tags);
-  }
-
-  constructor(logger: ServerFacade['log'], tags: string[]) {
+  constructor(logger: LoggerFactory, tags?: string[]) {
     this._logger = logger;
-    this._tags = tags;
+    this._tags = tags || [];
 
     /*
      * This shortcut provides maintenance convenience: Reporting code has been
      * using both .warn and .warning
      */
-    this.warn = this.warning.bind(this);
+    this.warning = this.warn.bind(this);
+  }
+
+  private getLogger(tags: string[]) {
+    return this._logger.get(...this._tags, ...tags);
   }
 
   public error(err: string | Error, tags: string[] = []) {
-    this._logger([...this._tags, ...tags, 'error'], err);
+    this.getLogger(tags).error(err);
   }
 
-  public warning(msg: string, tags: string[] = []) {
-    this._logger([...this._tags, ...tags, 'warning'], trimStr(msg));
+  public warn(msg: string, tags: string[] = []) {
+    this.getLogger(tags).warn(msg);
   }
 
   public debug(msg: string, tags: string[] = []) {
-    this._logger([...this._tags, ...tags, 'debug'], trimStr(msg));
+    this.getLogger(tags).debug(msg);
```

Comment:
I am finding that no debug messages are getting logged with the changes from this branch.

@joshdover you asked me to track this for you. I'll give it another run and file a bug if needed.

Reply:
Filed https://github.com/elastic/kibana/issues/55456